src/plugins/geometry_checker/qgsgeometrycheckersetuptab.cpp

```diff
@@ -54,16 +55,19 @@ QgsGeometryCheckerSetupTab::QgsGeometryCheckerSetupTab( QgisInterface *iface, QD
   mAbortButton = new QPushButton( tr( "Abort" ) );
   mRunButton->setEnabled( false );
 
-  QMap<QString, QString> filterFormatMap = QgsVectorFileWriter::supportedFiltersAndFormats();
-  for ( const QString &filter : filterFormatMap.keys() )
+  QgsSettings settings;
+
+  QMap<QString, QString> filterFormatMap = QgsVectorFileWriter::ogrDriverList();
+  ui.comboBoxOutputFormat->blockSignals( true );
+  for ( QMap< QString, QString>::const_iterator it = filterFormatMap.constBegin(); it != filterFormatMap.constEnd(); ++it )
   {
-    QString driverName = filterFormatMap.value( filter );
-    ui.comboBoxOutputFormat->addItem( driverName );
-    if ( driverName == QLatin1String( "ESRI Shapefile" ) )
-    {
-      ui.comboBoxOutputFormat->setCurrentIndex( ui.comboBoxOutputFormat->count() - 1 );
-    }
+    ui.comboBoxOutputFormat->addItem( it.key(), it.value() );
   }
+
+  QString format = settings.value( QStringLiteral( "UI/lastVectorFormat" ), "GPKG" ).toString();
+  ui.comboBoxOutputFormat->setCurrentIndex( ui.comboBoxOutputFormat->findData( format ) );
```

Comment:
Possibly worth using something like `std::max(0, ui.comboBoxOutputFormat->findData( format ))` since I suppose in theory it can happen that `lastVectorFormat` contains something which is not present in the list.